k8s/base.py

```diff
@@ -117,20 +118,38 @@ def list(cls, namespace="default"):
         return [cls.from_dict(item) for item in resp.json()[u"items"]]
 
     @classmethod
-    def watch_list(cls, namespace=None):
+    def _watch_url_from_base_resource_url(cls, base_resource_url, start_at_resource_version=None):
```

Comment:
Since the `client.get` call goes to `requests.request` in the end, passing any keyword arguments it gets, I think this whole thing could be replaced by simply passing a suitable `params` dictionary to the `client.get` call.
That would be much easier to work with, and easier to read than this method.